### PR TITLE
Improve language usage

### PR DESCRIPTION
**Pull request by Darhazer**
As a side effect, some cops will start detect `subject!` as well ;)

---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [ ] ~Added an entry to the [changelog](https://github.com/rubocop-rspec/rubocop-rspec/blob/master/CHANGELOG.md) if the new code introduces user-observable changes.~
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have created a new cop:

* [ ] Added the new cop to `config/default.yml`.
* [ ] The cop documents examples of good and bad code.
* [ ] The tests assert both that bad code is reported and that good code is not reported.

### CODE REVIEW

File: lib/rubocop/cop/rspec/let_before_examples.rb [hunk 1]
```diff
@@ -35,7 +35,8 @@ class LetBeforeExamples < Cop
 
         MSG = 'Move `let` before the examples in the group.'.freeze
 
-        def_node_matcher :let?, '(block (send nil? {:let :let!} ...) ...)'
+        def_node_matcher :let?, Helpers::ALL.block_pattern.freeze
```
Comment (bquorning) [inline lib/rubocop/cop/rspec/let_before_examples.rb]: You probably shouldn’t `freeze` here.

File: lib/rubocop/rspec/example_group.rb [hunk 2]
```diff
@@ -17,9 +17,7 @@ class ExampleGroup < Concept
       # @!method hook(node)
       #
       #   Detect if node is `before`, `after`, `around`
-      def_node_matcher :hook, <<-PATTERN
-        (block {$(send nil? #{Hooks::ALL.node_pattern_union} ...)} ...)
-      PATTERN
+      def_node_matcher :hook, Hooks::ALL.block_pattern
```
Comment (bquorning) [inline lib/rubocop/rspec/example_group.rb]: Please also rename the matcher to `hook?`.

**Comment by Darhazer** [comment]
Updated

File: lib/rubocop/rspec/example_group.rb [hunk 3]
```diff
@@ -14,14 +14,12 @@ class ExampleGroup < Concept
...
+      # @!method hook?(node)
       #
       #   Detect if node is `before`, `after`, `around`
-      def_node_matcher :hook, <<-PATTERN
-        (block {$(send nil? #{Hooks::ALL.node_pattern_union} ...)} ...)
-      PATTERN
+      def_node_matcher :hook?, Hooks::ALL.block_pattern
 
-      def_node_matcher :subject, Subject::ALL.block_pattern
+      def_node_matcher :subject?, Subject::ALL.block_pattern
```
Comment (bquorning) [inline lib/rubocop/rspec/example_group.rb]: ❤️

**Pull request merged**